### PR TITLE
DEP-2478: Changed text for PCS DA

### PR DESCRIPTION
**Pull request by tonymyers**
# DEP-(StoryNumberHere)

## Checklist PR Raiser
 - [x]  I've ensured code coverage has not decreased
 - [ ]  I've dealt with any new compilation warnings
 - [x]  I've ensured the team's coding standards have been met (TBC)?
 - [ ]  If relevant, I've created corresponding app-config-XXX changes?
 - [x]  If I've created new test data, I've ensured there is no real person's data (see https://confluence.tools.tax.service.gov.uk/display/CD/Test+Data+in+the+Open) 
 - [x]  I've considered the impact my changes have on the UI/Journey tests (please do not break them)
-  [ ]  If relevant, I've added links to associated PRs

## Checklist PR Reviewer
 - [x]  I've checked to ensure all relevant unit tests have been written
 - [ ]  I've checked to ensure the team's coding standards have been met
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  If I merge I will ensure I use squash and merge

### CODE REVIEW

File: app/assets/javascripts/controllers/CommonChatController.js [hunk 1]
```diff
@@ -181,7 +189,8 @@ export default class CommonChatController {
     closeChat() {
 
         if (document.body.contains(document.getElementById("postChatSurveyWrapper"))) {
-            this._sendPostChatSurveyWebchat(this.sdk).closePostChatSurvey(automaton, timestamp);
+            //this._sendPostChatSurveyWebchat(this.sdk).closePostChatSurvey(automaton, timestamp)
```
Comment (pratikrana06) [inline app/assets/javascripts/controllers/CommonChatController.js:192]: Can we get rid of commented code?

File: app/assets/javascripts/controllers/CommonChatController.js [hunk 2]
```diff
@@ -26,6 +28,16 @@ const webchatSurvey = {
     ]
 };
 
+const digitalAssistantSurvey = {
+    id: "13000303",
+    questions: [
+        { id: ["question1"], text: "Was the digital assistant useful?", freeform: false },
+        { id: ["question2"], text: "How could we improve it?", freeform: false },
+        { id: ["question3"], text: "If you had not used the chatbot today, how else would you have contacted us?", freeform: false },
+        { id: ["question4"], text: "Provide other contact options?", freeform: false }
```
Comment (pratikrana06) [inline app/assets/javascripts/controllers/CommonChatController.js:37]: Can we please remove ? at the end?

File: app/assets/javascripts/controllers/CommonChatController.js [hunk 3]
```diff
@@ -230,7 +243,7 @@ export default class CommonChatController {
             console.log("************************************")
             console.log("******* chat is in progress ********")
             console.log("************************************")
-            //            setTimeout(() => this._launchChat(), 2000);
+                //            setTimeout(() => this._launchChat(), 2000);
```
Comment (pratikrana06) [inline app/assets/javascripts/controllers/CommonChatController.js:246]: Can we get rid of commented code?

File: app/assets/javascripts/controllers/CommonChatController.js [hunk 4]
```diff
@@ -274,8 +287,10 @@ export default class CommonChatController {
 
     onConfirmEndChat() {
         this._moveToClosingState();
-        this._sendPostChatSurveyWebchat(this.sdk).beginPostChatSurvey(webchatSurvey, automaton, timestamp);
-        this.container.showPage(new PostChatSurveyWebchat((page) => this.onPostChatSurveyWebchatSubmitted(page)));
+        //this._sendPostChatSurveyWebchat(this.sdk).beginPostChatSurvey(webchatSurvey, automaton, timestamp)
```
Comment (pratikrana06) [inline app/assets/javascripts/controllers/CommonChatController.js:290]: Can we get rid of commented code?

File: app/assets/javascripts/controllers/CommonChatController.js [hunk 5]
```diff
@@ -298,4 +313,21 @@ export default class CommonChatController {
...
+        };
+
+        var surveyWithAnswers = Object.assign(answers, digitalAssistantSurvey);
+
+        this._sendPostChatSurveyDigitalAssistant(this.sdk).submitPostChatSurvey(surveyWithAnswers, automaton, timestamp);
+        surveyPage.detach();
+        this.showEndChatPage(true);
+    }
+
 };
```
Comment (pratikrana06) [inline app/assets/javascripts/controllers/CommonChatController.js:333]: Can we add New line at the end of the file please?

File: app/assets/javascripts/services/PostChatSurveyDigitalAssistantService.js [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        };
+        console.log("===== closePostChatSurvey =====");
+
+        try {
+            this.sdk.logEventToDW({ eventList: [endedEvent] });
+        } catch (e) {
+            console.error("!!!! logEventToDW got exception: ", e);
+        }
+    }
+}
```
Comment (pratikrana06) [inline app/assets/javascripts/services/PostChatSurveyDigitalAssistantService.js:246]: Can we add New line at the end of the file please?

File: app/assets/javascripts/views/postChatSurvey/PostChatSurveyDigitalAssistant.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+           if ($(this).val() != 'Other') {
+                document.getElementById("q4-").value = "";
+           }
+        });
+    }
+
+    detach() {
+        this.container.removeChild(this.wrapper)
+    }
+}
```
Comment (pratikrana06) [inline app/assets/javascripts/views/postChatSurvey/PostChatSurveyDigitalAssistant.js:111]: Can we add New line at the end of the file please?

**Review by pratikrana06 — COMMENTED** [review]
Review comments added

File: test/uk/gov/hmrc/digitalengagementplatformskin/javascripts/services/PostChatSurveyDigitalAssistantService.spec.js [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+    ]
+};
+
+const surveyWithoutOption = {
+    id: 123456,
+    questions: [
+        { id: "q1", text: "Was the digital assistant useful?", freeform: false },
+        { id: "q2", text: "How could we improve it?", freeform: false },
+        { id: "q3", text: "If you had not used the chatbot today, how else would you have contacted us?", freeform: false },
+        { id: "q4", text: "Provide other contact options?", freeform: false }
```
Comment (pratikrana06) [inline test/uk/gov/hmrc/digitalengagementplatformskin/javascripts/services/PostChatSurveyDigitalAssistantService.spec.js:48]: Can we please remove ? at the end?

File: app/assets/javascripts/controllers/CommonChatController.js (hunk 1 above)
Comment (tonymyers) [reply app/assets/javascripts/controllers/CommonChatController.js:192]: No, will make DEP-2480 easier as we need both methods for that ticket

File: app/assets/javascripts/controllers/CommonChatController.js (hunk 3 above)
Comment (tonymyers) [reply app/assets/javascripts/controllers/CommonChatController.js:246]: Not for the minute if that's ok

File: app/assets/javascripts/controllers/CommonChatController.js (hunk 4 above)
Comment (tonymyers) [reply app/assets/javascripts/controllers/CommonChatController.js:290]: No, will make DEP-2480 easier as we need both methods for that ticket

**Review by pratikrana06 — APPROVED** [review]
Approved

File: app/assets/javascripts/controllers/CommonChatController.js (hunk 2 above)
Comment (tonymyers) [reply app/assets/javascripts/controllers/CommonChatController.js]: Done

File: app/assets/javascripts/controllers/CommonChatController.js (hunk 5 above)
Comment (tonymyers) [reply app/assets/javascripts/controllers/CommonChatController.js]: Done

File: app/assets/javascripts/services/PostChatSurveyDigitalAssistantService.js (hunk 6 above)
Comment (tonymyers) [reply app/assets/javascripts/services/PostChatSurveyDigitalAssistantService.js]: Done

File: app/assets/javascripts/views/postChatSurvey/PostChatSurveyDigitalAssistant.js (hunk 7 above)
Comment (tonymyers) [reply app/assets/javascripts/views/postChatSurvey/PostChatSurveyDigitalAssistant.js]: Done

File: test/uk/gov/hmrc/digitalengagementplatformskin/javascripts/services/PostChatSurveyDigitalAssistantService.spec.js (hunk 8 above)
Comment (tonymyers) [reply test/uk/gov/hmrc/digitalengagementplatformskin/javascripts/services/PostChatSurveyDigitalAssistantService.spec.js]: Done

**Pull request merged**